packages/lib/src/__tests__/user-query.test.ts

```diff
@@ -0,0 +1,247 @@
+import inquirer from 'inquirer'
+import { askForInteger, askForNumber, askForRequiredString, askForString, numberTransformer, ValidateFunction } from '../user-query'
+
+
+describe('user-query', () => {
+	const promptSpy = jest.spyOn(inquirer, 'prompt')
+
+	afterEach(() => {
+		jest.clearAllMocks()
+	})
+
+	describe('numberTransformer', () => {
+		it.each(['', 'input value', '7'])('returns input value unchanged when not final', value => {
+			expect(numberTransformer(value, { value }, { isFinal: false })).toBe(value)
+		})
+
+		it.each(['input value', '7'])('returns input value unchanged when not empty', value => {
+			expect(numberTransformer(value, { value }, { isFinal: true })).toBe(value)
+		})
+
+		it('returns "none" when isFinal and input value is empty', () => {
+			expect(numberTransformer('', { value: '' }, { isFinal: true })).toBe('none')
+		})
+	})
+
+	describe('askForString', () => {
+		it('ask user correct question', async () => {
+			promptSpy.mockResolvedValue({ value: 'entered value' })
+
+			const result = await askForString('prompt message')
+
+			expect(result).toBe('entered value')
+			expect(promptSpy).toHaveBeenCalledTimes(1)
+			expect(promptSpy).toHaveBeenCalledWith({
+				type: 'input',
+				name: 'value',
+				message: 'prompt message',
+			})
+		})
+
+		it('returns nothing entered as undefined', async () => {
+			promptSpy.mockResolvedValue({ value: '' })
+
+			const result = await askForString('prompt message')
+
+			expect(result).toBe(undefined)
+			expect(promptSpy).toHaveBeenCalledTimes(1)
+			expect(promptSpy).toHaveBeenCalledWith({
+				type: 'input',
+				name: 'value',
+				message: 'prompt message',
+			})
+		})
+	})
+
+	describe('askForRequiredString', () => {
+		it('requires input', async () => {
+			promptSpy.mockResolvedValue({ value: 'entered value' })
+
+			const result = await askForRequiredString('prompt message')
+
+			expect(result).toBe('entered value')
+			expect(promptSpy).toHaveBeenCalledTimes(1)
+			expect(promptSpy).toHaveBeenCalledWith({
+				type: 'input',
+				name: 'value',
+				message: 'prompt message',
+				validate: expect.any(Function),
+			})
+
+			const validateFunction = (promptSpy.mock.calls[0][0] as { validate: ValidateFunction }).validate
+
+			expect(validateFunction('')).toBe('value is required')
+			expect(validateFunction('a')).toBe(true)
+		})
+	})
+
+	describe('askForInteger', () => {
+		it.each`input   | expected
+				${'-1'} | ${-1}
+				${'0'}  | ${0}
+				${'7'}  | ${7}
+				${''}   | ${undefined}`('returns number entered', async ({ input, expected }) => {
```

Comment:
The formatting is a little off here.

Reply:
I think in this case, that is just an artifact of github using 8 spaces for a tab.

Reply:
Yep, that's it.